extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/transaction/MongoTransactional.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.mongodb.panache.transaction;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.interceptor.InterceptorBinding;
+
+/**
+ * Indicate that this methods must be executed inside MongoDB transaction boundaries.
+ *
+ * See https://docs.mongodb.com/manual/core/transactions/
+ */
+@Inherited
+@InterceptorBinding
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MongoTransactional {
+
+    /**
+     * Read preference for this transaction defined as a String.
+     * ReadPreference#valueOf(String) will be used to retrieve the write concern.
+     * If not specified, the default from the underlying MongoClient will be used.
```

Comment:
This JavaDoc would be more useful if you were listing the various read preference options and what they mean. Same for the other ones below